### ISSUE TITLE
We should also test doc values

### ISSUE THREAD

**Issue by mikemccand** [Tony-X/search-benchmark-game#34]
Doc values are an important part of most search applications, and we're not testing them today.

**Comment by fulmicoton** [comment]
That would be extremely interesting. This is one of the piece where tantivy diverged quite drastically from lucene too.
Our codec is similar but is tweaked for random access, and we do random access on it. I would love ot see some comparison.

The SIMD stuff discussed in the other ticket only kicks in for range queries.

**Comment by mikemccand** [comment]
> The SIMD stuff discussed in the other ticket only kicks in for range queries.

OK got it.

> That would be extremely interesting. This is one of the piece where tantivy diverged quite drastically from lucene too.
> Our codec is similar but is tweaked for random access, and we do random access on it. I would love ot see some comparison.

Ahh that is interesting!  Lucene used to be (long ago) random accessed, and then moved to iterator only, which was problematic e.g. for KNN search.

Hmm, for the default BM25 rank, is Tantivy using any doc values, e.g. maybe field or doc length (norms)?  Lucene does store norms as doc values, so we are testing that in the search benchmark now.  I'll open a spinoff issue to also test non-relevance sort (though that, too, will likely be another doc values field).

**Comment by fulmicoton** [comment]
Yes... But we did not have enough time to do proper research on whether this was a good idea or not, so I am super interested in whatever benchmark you can come up with.

If you want to experiment with random access docvalues, I am happy to share details about our codec. I really need to write a blog post about that.

> Hmm, for the default BM25 rank, is Tantivy using any doc values, e.g. maybe field or doc length (norms)? Lucene does store norms as doc values, so we are testing that in the search benchmark now. I'll open a spinoff issue to also test non-relevance sort (though that, too, will likely be another doc values field).

In tantivy, Fieldnorms are stored over a single byte, using the same "scale" as Lucene.
The fieldnorms are not exactly docvalues. They will always take one byte per doc, even if the field is absent from the document. In that case, it will get the fieldnorm 0.

We have a way to index stuff in a schemaless manner -mostly useful in quickwit-, but in that case, you don't have any fieldnorms.
